python/pyspark/sql/column.py

```diff
@@ -250,11 +250,39 @@ def __iter__(self):
         raise TypeError("Column is not iterable")
 
     # string methods
+    _rlike_doc = """ Return a Boolean :class:`Column` based on a regex match.\n
+                :param other: an extended regex expression\n
+
+                >>> df.filter( df.name.rlike('ice$') ).collect()
+                [Row(name=u'Alice', age=1)]
+                """
+    _like_doc = """ Return a Boolean :class:`Column` based on a SQL LIKE match.\n
+               :param other: a SQL LIKE pattern\n
+               See :func:`pyspark.sql.Column.rlike` for a regex version\n
+
+               >>> df.filter( df.name.like('Al%') ).collect()
+               [Row(name=u'Alice', age=1)]
+                """
+    _startswith_doc = ''' Return a Boolean :class:`Column` based on a string match.\n
```

Comment:
Are [these the Scala docs](http://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.sql.Column@rlike(literal:String):org.apache.spark.sql.Column) you're referring to? It looks like the Scala Column docs are missing examples for a couple more functions that are documented in python (substr, isin). I could expand the ticket / PR to include Scala examples for all of these functions. I also noticed isNull and isNotNull are not document either.